openssl/goopenssl.h

```diff
@@ -1054,19 +1062,43 @@ DEFINEFUNC(int, EVP_PKEY_keygen_init, (GO_EVP_PKEY_CTX *ctx), (ctx))
 DEFINEFUNC(int, EVP_PKEY_keygen, (GO_EVP_PKEY_CTX *ctx, GO_EVP_PKEY **ppkey), (ctx, ppkey))
 
 #if OPENSSL_VERSION_NUMBER >= 0x30000000L
-enum {
-	GO_POINT_CONVERSION_UNCOMPRESSED = POINT_CONVERSION_UNCOMPRESSED,
-};
+typedef OSSL_PARAM GO_OSSL_PARAM;
 
 enum {
 	GO_OSSL_PARAM_UNMODIFIED = OSSL_PARAM_UNMODIFIED,
 	GO_OSSL_PARAM_UNSIGNED_INTEGER = OSSL_PARAM_UNSIGNED_INTEGER,
 	GO_OSSL_PARAM_UTF8_STRING = OSSL_PARAM_UTF8_STRING,
 	GO_OSSL_PARAM_OCTET_STRING = OSSL_PARAM_OCTET_STRING
 };
+enum {
+	GO_POINT_CONVERSION_UNCOMPRESSED = POINT_CONVERSION_UNCOMPRESSED,
+};
+#endif
 
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L
+#if OPENSSL_VERSION_NUMBER < 0x30000000L
+typedef struct ossl_param_st GO_OSSL_PARAM;
+struct ossl_param_st {
```

Comment:
It looks a bit cumbersome to expose the OpenSSL 3-only interface to make `params.go` compile, though it is practically unused when an earlier version of OpenSSL is detected at run time. Perhaps, we could port the code using OSSL_PARAM into C and only expose minimal set of functions?

Reply:
I'm not sure if it is feasible but perhaps we could use a build tag for conditional compilation, e.g., openssl10, openssl11, and openssl3, so that
- `newECDHPkey`  is defined differently depending on the build tag
- `params.go` (used by `newECDHPkey` implementation for OpenSSL 3) is only compiled with `+build openssl3`

Reply:
I think a build tag could work, however that could complicate the RPM builds a bit so I would like @dbenoit17 input on that.